### PR TITLE
Full inventory report with obs sanity check

### LINKED ISSUE #229: Obs database inventory
Write python based tools to generate an inventory of the database. 

**Input:**
- A list of obstype (`adt`, `sst`, `sss`, `icec`, ...), or maybe we simply get the full list from the DB
- path to an r2d2 database

**Output:** 
- A complete detailed report of the list of platforms/instruments available from the database for the obs type specified in the input, along with the date range of availability   

That could be organized in "barplots" maybe? Something like that:

inventory-obsdb

### PR DESCRIPTION
**Pull request by mjhossen**
# Description

## Overview of what this PR does
Checking and reviewing the scripts written for full inventory report and observation sanity check.

## details here

 I pushed two scripts: godas_obssanitycheck.py and godas_inventory_plot.py

The python script "godas_inventory_plot.py" provides full inventory report based on data availability in the R2D2 database.  The code can be used for P1D and PT10M (SST 10 min). The command is 
For P1D: python godas_inventory_plot.py -s 20150101 -e 20220101 -y obs_inventory_p1d.yaml 
For PT10M:
python godas_inventory_plot.py -s 201501010000 -e 202201010000 -y obs_inventory_pt10m.yaml 

The "godas_obssanitycheck.py" script is useful to find more details, such as obs_count, min/max of ObsValue or ObsError and missing dates. This can be used for smaller range of time period where some data are missing so that the missing observation files can be filled up in the database. Again, this works for both P1D and PT10M. The command is 
for P1D: godas_obssanitycheck.py -t adt -p c2 j2 j3 sa 3a -s 20150101 -e 20150401 --step P1D 
for PT10M: godas_obssanitycheck.py -t sst -p noaa18 noaa19 -s 201501010000 -e 201501050000 --step PT10M 

- fixes #229

### CODE REVIEW

**Comment by guillaumevernieres** [comment]
1 - Title of the PR: Make an effort and choose an appropriate title, not just the default (branch name)
2 - We need a clear description of the work that was done in you PR description
3 - This PR should be closing an issue when merged: so at the end of the descriptioon you should add somehting like:
- closes #999

@mjhossen , click edit on you PR description and make the changes suggested in 1-2 and 3 above.

**Comment by guillaumevernieres** [comment]
That's better @mjhossen . A few more presentation comments:

# Make use of mark down. 
Knowing basic markdown command will make your description easier to read. Google is your friend if you need to learn the basics.

Example of what I think is a more readable presentation (copy paste of what's in your description):

[...]
The python script "godas_inventory_plot.py" provides full inventory report based on data availability in the R2D2 database.
The code can be used for P1D and PT10M (SST 10 min). 
The command is
```console
python godas_inventory_plot.py -s 20150101 -e 20220101 -y obs_inventory_p1d.yaml     # For P1D 
```
and
```console
python godas_inventory_plot.py -s 201501010000 -e 202201010000 -y obs_inventory_pt10m.yaml # For PT10M:
```

# Don't state the obvious
Things like close the issue after merging or what you currently wrote: `This PR should be closed after merging` are not useful. 
It's not for you to decide, it's up to the reviewers, so no mention of close/merge this or that. Evidently, the point of a PR is to eventually merge code.

Hope this helps.

**Comment by mjhossen** [comment]
Removed "This PR should be closed after merging"

File: obsdiag/godas_obssanitycheck.py [hunk 1]
```diff
@@ -6,18 +6,50 @@
...
 import matplotlib.dates as mdates
 import matplotlib.gridspec as gridspec
+import sys
+from glob import glob
 
 class ObsSanityCheck:
-    def __init__(self,args=None, filename=None, folder=None):
-        self.fname = filename
-        self.folder = folder
+    def __init__(self,args=None):
```
Comment (guillaumevernieres) [inline obsdiag/godas_obssanitycheck.py:13]: it doesn't look like args is optional, why set it to `None`?

**Review by guillaumevernieres — APPROVED** [review]
Thanks for doing this @mjhossen , going to merge as is for now.

**Pull request merged**